sklearn/preprocessing/_encoders.py

```diff
@@ -204,14 +231,47 @@ class OneHotEncoder(_BaseEncoder):
     dtype : number type, default=np.float
         Desired dtype of output.
 
-    handle_unknown : {'error', 'ignore'}, default='error'
+    handle_unknown : {'error', 'ignore', 'auto'}, default='error'
         Whether to raise an error or ignore if an unknown categorical feature
         is present during transform (default is to raise). When this parameter
         is set to 'ignore' and an unknown category is encountered during
         transform, the resulting one-hot encoded columns for this feature
         will be all zeros. In the inverse transform, an unknown category
         will be denoted as None.
 
+        When this parameter is set to 'auto' and an unknown category is
+        encountered in transform:
+
+        1. If there was no infrequent category during training, the resulting
+        one-hot encoded columns for this feature will be be all zeros. In
+        the inverse transform, an unknown category will be denoted as None.
+        2. If there is an infrequent category during training, the unknown
+        category will be considered infrequent. In the inverse transform,
+        an unknown category will be the most frequent infrequent category
+
+        .. versionadded:: 0.23
+            'auto' was added to automatically handle unknown categories
+
+        .. deprecated:: 0.23
+            'ignore' is deprecated in favor of 'auto'
+
+    min_frequency : int or float, default=1
+        Specifics the categories to be considered infrequent.
+
+        1. If int, categories with a cardinality smaller will be considered
+        infrequent.
+        2. If float, categories with a cardinality smaller than this fraction
+        of the total number of samples will be considered infrequent.
+
+        .. versionadded:: 0.23
+
+    max_levels : int, default=None
+        Specifies the categories to be considered infrequent. Sets an upper
```

Comment:
doesn't maxlevels specify the number of **frequent** categories instead?

Reply:
In this implementation it specifies the total number of categories (including the infrequent). I have updated the docs to reflect this.